### PR TITLE
fix: GST Tax Type in Journal Entry

### LINKED ISSUE #2300: GST Tax Type in Journal Entry

### PR DESCRIPTION
**Pull request by Sanket322**
closes : #2300

### CODE REVIEW

File: india_compliance/gst_india/doctype/gstr_3b_report/gstr_3b_report.py [hunk 1]
```diff
@@ -187,18 +186,25 @@ def update_itc_reversal_from_journal_entry(self):
         )
 
         net_itc = self.report_dict["itc_elg"]["itc_net"]
+        account_map = {
+            "sgst": "samt",
+            "cess": "csamt",
+            "cgst": "camt",
+            "igst": "iamt",
+        }
```
Comment (ljain112) [inline india_compliance/gst_india/doctype/gstr_3b_report/gstr_3b_report.py:206]: .

File: india_compliance/gst_india/overrides/journal_entry.py [hunk 2]
```diff
@@ -5,6 +5,20 @@
...
+    if not doc.accounts:
+        return
+
+    gst_tax_account_map = get_gst_account_gst_tax_type_map()
+    if not gst_tax_account_map:
+        frappe.throw(_("Please set GST Accounts in GST Settings"))
+
+    for tax in doc.accounts:
+        # Setting as None if not GST Account
+        tax.gst_tax_type = gst_tax_account_map.get(tax.account)
```
Comment (ljain112) [inline india_compliance/gst_india/overrides/journal_entry.py:20]: .

File: india_compliance/patches.txt [hunk 3]
```diff
@@ -56,4 +56,5 @@ india_compliance.patches.v14.unset_inward_supply_link_for_cancelled_purchase
 india_compliance.patches.v14.delete_not_generated_gstr_import_log
 india_compliance.patches.v14.enable_sales_through_ecommerce_operator
 india_compliance.patches.post_install.set_gst_tax_type
+india_compliance.patches.post_install.set_gst_tax_type_in_general_entry
```
Comment (ljain112) [inline india_compliance/patches.txt]: .

File: india_compliance/patches/post_install/set_gst_tax_type_in_general_entry.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import frappe
+from frappe.query_builder import Case
+
+from india_compliance.gst_india.utils import get_gst_account_gst_tax_type_map
+
+
+def execute():
+    gst_tax_type_account_map = get_gst_account_gst_tax_type_map()
+    if not gst_tax_type_account_map:
```
Comment (ljain112) [inline india_compliance/patches/post_install/set_gst_tax_type_in_general_entry.py]: .

**Review by vorasmit — APPROVED** [review]

**Pull request merged**